README.md

```diff
@@ -2,49 +2,211 @@
+looking at them, but here's a quickstart guide.
+
+### Basic Usage
+
+To load `cl-syslog`, use Quicklisp or ASDF as usual:
+
+```
+> (ql:quickload :cl-syslog)`
+;; or, if ASDF can already see it:
+> (asdf:load-system :cl-syslog)
+```
+
+Priorities and facilities are stated by their keyword name. For
+instance, the priority `:warning` on facility `:user`. All of these
+keywords are housed in the variables `*priorities*` and
+`*variables*`. Should you provide invalid facilities or priorities,
+the conditions `cl-syslog:invalid-facility` and
+`cl-syslog:invalid-priority` will be signaled.
+
+To log to your local syslog daemon, use `cl-syslog:log` (note that
+this is shadowed, and is obviously not the same as `cl:log`!):
+
+```
+> (cl-syslog:log "MyApp" ':user ':warning "Low on memory.")
+```
+
+Then look in your `/var/log/messages` or other location if you have
+tweaked your `/etc/syslog.conf`. On macOS, you can use the Console
+application.
+
+### RFC 5424 Support
+
+RFC 5424 is supported. (You can check if your CL-SYSLOG installation
+support it by checking if `cl-syslog::rfc5424` is present in your
+`*features*`.)
+
+The first thing you'll need to do is create an instance of the
+`rfc5424-logger` class, filling out as many details as you
+please. You're encouraged to fill out as many slots as you can so that
+log messages contain more information.
+
+Of particular interest is the `:log-writer` initarg. This controls how
+log messages get processed in the end. The default log writer will use
+a function wrapping `cl-syslog:log`. Should you instead be interested
+in writing to standard output, you can supply
+`(cl-syslog:stream-log-writer)` as an argument, like so:
+
+```
+(defparameter *logger*
+ (make-instance 'cl-syslog:rfc5424-logger
+                ;; optional
+                :facility ':local0
+                :app-name "MyApp"
+                :hostname "computer.local"
+                :log-writer (cl-syslog:stream-log-writer)))
+```
+
+There are a few options for log writers, including a UDP log writer
+for remote RFC-compliant logging. You can always support your own
+`lambda` function, too.
+
+Once you have a logger, the simplest way to create a compliant message
+is to use `cl-syslog:format-log`:
+
+```
+> (cl-syslog:format-log *logger* ':warning "Low on memory.")
+<132>1 2018-12-13T22:17:56Z computer.local MyApp 69840 - - Low on memory.
+```
+
+RFC 5424 has support for sending structured data within the log
+message. It's a somewhat complicated arrangement, in that the
+structured data has to be defined and "agreed upon". Specifically,
+there are a set of IETF-approved structured data. They're identified
+by "structured data IDs", and the standard IDs are `timeQuality`,
+`origin`, and `meta`. Each of these have different structured data
+parameters.
+
+Personal structured data IDs must have some name, followed by `@`,
+followed by a usually 5 digit integer. For instance, `mycompany@0001`
```

Comment:
That is not 5 digits. Maybe say "by a usually (but not necessarily) 5 digit integer"